lib/http2/connection.c

```diff
@@ -681,8 +691,7 @@ static int handle_headers_frame(h2o_http2_conn_t *conn, h2o_http2_frame_t *frame
             if ((frame->flags & H2O_HTTP2_FRAME_FLAG_END_HEADERS) == 0)
                 goto PREPARE_FOR_CONTINUATION;
             return handle_trailing_headers(conn, stream, payload.headers, payload.headers_len, err_desc);
-        } else if (!stream || stream->state != H2O_HTTP2_STREAM_STATE_IDLE) {
```

Comment:
Why isn't this needed any more? This check was added so that we don't send an error back if the stream was opened by a `PRIORITY` frame and thus is in the IDLE state.

Reply:
@deweerdt 
I think You mean https://github.com/h2o/h2o/pull/1043, but after that change, `max_open` came to be not increased by PRIORITY frame (https://github.com/h2o/h2o/pull/1136, https://github.com/h2o/h2o/pull/1138). By that fact, a stream with prior PRIORITY frame must have the id which is lesser or equal than max_open, so I think this condition is no longer needed. Please correct me if i'm wrong.

Reply:
Ah, makes sense. Thank you.